sdl_viewer/src/main.rs

```diff
@@ -210,57 +211,72 @@ impl NodeView {
 // Keeps track of the nodes that were requested in-order and loads then one by one on request.
 struct NodeViewContainer {
     node_views: HashMap<octree::NodeId, NodeView>,
-    queue: VecDeque<octree::NodeId>,
     queued: HashSet<octree::NodeId>,
 }
 
 impl NodeViewContainer {
     fn new() -> Self {
         NodeViewContainer {
             node_views: HashMap::new(),
-            queue: VecDeque::new(),
             queued: HashSet::new(),
         }
     }
 
-    // Loads the next most-important nodes data. Returns false if there are no more nodes queued
-    // for loading.
-    fn load_next_node(&mut self, octree: &octree::Octree, program: &GlProgram) -> bool {
-        // We always request nodes at full resolution (i.e. not subsampled by the backend), because
-        // we can just as effectively subsample the number of points we draw in the client.
-        const ALL_POINTS_LOD: i32 = 1;
-        if let Some(node_id) = self.queue.pop_front() {
+    // Returns the 'NodeView' for 'node_id' if it is already loaded, otherwise returns None, but
+    // registered the node for loading in the worker thread
+    fn get(&mut self, node_id: &octree::NodeId, program: &GlProgram, sender: &mut Sender<octree::NodeId>, receiver: &mut Receiver<(octree::NodeId, octree::NodeData)>) -> Option<&NodeView> {
```

Comment:
I suggest a change of design here: This struct should start the background thread in its new() method. It can then also hold on to the sender and reveiver. We can also get rid of the FromDiscLoader struct by inlining its run method into the closure that the thread takes.